### PR TITLE
Add Teams Management

### PR DESCRIPTION
**Pull request by fivetran-aleksandrboldyrev**

### CODE REVIEW

**Review by beevital — COMMENTED** [review]
I'm not 100% sure about connector <-> team and group <-> team memberships, but for team users we should definitely have the same logic as we have for group users.

Maybe it's ok for now to have granular resource that represents relation of connector and team, or group and team.
We should make some analysis about amount ratio: for example I bet that most customers have much more connectors that teams, so it makes sense to group connector memberships by team like 
```
resource "fivetran_team_connectors" "team_connectors" {
   team_id = "some_team_id"
   connector {
      id = "connector1_id"
      role = "team_role_in_connector1"
   }
   connector {
      id = "connector2_id"
      role = "team_role_in_connector2"
   }
   ... etc
}
```

But for groups - most users have 1 group, maybe 2 groups (one for production and one as a sandbox). So I'd prefer to have the following resource:
```
resource "fivetran_team_groups" "team_groups" {
   team_id = "some_team_id"
   group {
      id = "group1_id"
      role = "team_role_in_group1"
   }
   group {
      id = "group2_id"
      role = "team_role_in_group2"
   }
   ... etc
}
```

File: fivetran/resource_team_user_membership.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            Description: "The User unique identifier",
+        },
+        "role": {
+            Type:        schema.TypeString,
+            Required:    !datasource,
+            Computed:    datasource,
+            Description: "The team's role that links the team and the User",
+        },
+    }
+}
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/resource_team_user_membership.go]: Done

File: fivetran/resource_team_group_membership.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            Computed:    datasource,
+            Description: "The team's role that links the team and the Group",
+        },
+        "created_at": {
+            Type:        schema.TypeString,
+            Computed:    true,
+            Description: "The date and time the membership was created",
+        },
+    }
+}
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/resource_team_group_membership.go]: Done

File: fivetran/resource_team_connector_membership.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            Computed:    datasource,
+            Description: "The team's role that links the team and the connector",
+        },
+        "created_at": {
+            Type:        schema.TypeString,
+            Computed:    true,
+            Description: "The date and time the membership was created",
+        },
+    }
+}
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/resource_team_connector_membership.go]: Done

File: fivetran/data_source_team_user_memberships.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: getTeamUserMembershipSchema(true),
+				},
+			},
+		},
+	}
+}
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/data_source_team_user_memberships.go]: Done

File: fivetran/data_source_team_group_memberships.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: getTeamGroupMembershipSchema(true),
+				},
+			},
+		},
+	}
+}
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/data_source_team_group_memberships.go]: Done

File: fivetran/data_source_team_connector_memberships.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: getTeamConnectorMembershipSchema(true),
+				},
+			},
+		},
+	}
+}
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/data_source_team_connector_memberships.go]: Done

File: fivetran/resource_team.go [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Required:    !datasource,
+			Computed:    datasource,
+			Description: "The name of the team within your account.",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Required:    !datasource,
```
[the remark this answers is not in the document]
Comment (fivetran-aleksandrboldyrev) [reply fivetran/resource_team.go]: Fixed

**Review by beevital — APPROVED** [review]
LGTM. Have minor changes requests.

I also have a suggestion for further refactoring: all memberships resources have exact the same logic in update methods - the difference only in methods that are called on membership CRUD. 
We could generify these methods, and use the same method also in group-users resource.

**Pull request merged**